### PR TITLE
Mitigate data corruption effects like constant UI refresh

### PR DESCRIPTION
**Pull request by ggalmazor**
This PR acts on the repeated refresh problem of the UI due to data corruption related issues in the database. The changes include:

- Validity checks on `Form` that detect missing pieces from related `_form_info_x_y_z` tables
- Log traces for unhandled exceptions in `FormServiceImpl`

This is related to #114

### CODE REVIEW

File: src/main/java/org/opendatakit/aggregate/constants/ErrorConsts.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
  */
 public final class ErrorConsts {
 
+  public static final String POSSIBLE_CORRUPTION = "Encountered an unknown problem while reading the database. This often is a sign of database corruption. Please get the server logs and file an issue";
```
Comment (ggalmazor) [inline src/main/java/org/opendatakit/aggregate/constants/ErrorConsts.java]: This is a catch-all error for unhandled exceptions

File: src/main/java/org/opendatakit/aggregate/form/Form.java [hunk 2]
```diff
@@ -713,5 +713,8 @@ public String getUri() {
     return infoRow.getUri();
   }
 
-
+  @Override
+  public boolean isValid() {
```
Comment (ggalmazor) [inline src/main/java/org/opendatakit/aggregate/form/Form.java]: This definition of `isValid()` would grow with more checks as we go.

File: src/main/java/org/opendatakit/aggregate/form/FormDefinition.java [hunk 3]
```diff
@@ -135,9 +136,13 @@ private static final SubmissionAssociationTable getSubmissionAssociation(String
 		    		sa = st;
 		    	}
 		    }
-		}
+			}
 	    return sa;
-	}
+		} catch (Throwable t) {
```
Comment (ggalmazor) [inline src/main/java/org/opendatakit/aggregate/form/FormDefinition.java]: `st.getIsPersistenceModelComplete()` in line 125 will throw an `NPE` when that field is `null`. This new `try` block ensures that this exception doesn't scale. Calling code at `FormDefinition` can handle `null` `SubmissionAssociationTable` values.

File: src/main/java/org/opendatakit/aggregate/constants/ErrorConsts.java (hunk 1 above)
Comment (lognaturel) [reply src/main/java/org/opendatakit/aggregate/constants/ErrorConsts.java]: How about adding https://github.com/opendatakit/aggregate/issues after add an issue?

**Comment by ggalmazor** [comment]
Please, don't merge this PR. I'll produce a new one without the mess in git history.

**Comment by ggalmazor** [comment]
This PR continues in #154

**Pull request closed without merging**